pkg/operator/edgefs/cluster/utils.go

```diff
@@ -0,0 +1,313 @@
+		}
+	}
+	validNodes := k8sutil.GetValidNodes(c.Spec.Storage.Nodes, c.context.Clientset, edgefsv1alpha1.GetTargetPlacement(c.Spec.Placement))
+	c.Spec.Storage.Nodes = validNodes
+	return validNodes, nil
+}
+
+func (c *cluster) createDeploymentConfig(nodes []rookalpha.Node, resurrect bool) (edgefsv1alpha1.ClusterDeploymentConfig, error) {
+
+	deploymentConfig := edgefsv1alpha1.ClusterDeploymentConfig{DevConfig: make(map[string]edgefsv1alpha1.DevicesConfig, 0)}
+	//Fill deploymentConfig devices struct
+	for _, node := range nodes {
+		n := c.resolveNode(node.Name)
+
+		if n == nil {
+			return deploymentConfig, fmt.Errorf("node %s did not resolve to start target", node.Name)
+		}
+
+		devicesConfig := edgefsv1alpha1.DevicesConfig{}
+		devicesConfig.Rtrd.Devices = make([]edgefsv1alpha1.RTDevice, 0)
+		devicesConfig.Rtlfs.Devices = make([]edgefsv1alpha1.RtlfsDevice, 0)
+
+		// if node labeled as gateway then return empty devices and skip RTDevices detection
+		if c.isGatewayLabeledNode(c.context.Clientset, node.Name) {
+			devicesConfig.IsGatewayNode = true
+			deploymentConfig.DevConfig[node.Name] = devicesConfig
+			continue
+		}
+
+		rookSystemNS := os.Getenv(k8sutil.PodNamespaceEnvVar)
+		nodeDevices, _ := discover.ListDevices(c.context, rookSystemNS, n.Name)
+
+		availDevs, deviceErr := discover.GetAvailableDevices(c.context, n.Name, c.Namespace,
+			n.Devices, n.Selection.DeviceFilter, n.Selection.GetUseAllDevices())
+
+		if deviceErr != nil {
+			// Devices were specified but we couldn't find any.
+			// User needs to fix CRD.
+			return deploymentConfig, fmt.Errorf("failed to get devices for node %s cluster %s: %v",
+				n.Name, c.Namespace, deviceErr)
+		}
+
+		// selects Disks from availDevs and translate to RTDevices
+		availDisks := []sys.LocalDisk{}
+		for _, dev := range availDevs {
+			for _, disk := range nodeDevices[n.Name] {
+				if disk.Name == dev.Name {
+					availDisks = append(availDisks, disk)
+				}
+			}
+		}
+
+		storeConfig := config.ToStoreConfig(n.Config)
+		rtDevices, err := target.GetRTDevices(availDisks, &storeConfig)
+		if err != nil {
+			logger.Warningf("Can't get rtDevices for node %s due %v", n.Name, err)
+			rtDevices = make([]edgefsv1alpha1.RTDevice, 0)
+		}
+
+		devicesConfig.Rtrd.Devices = rtDevices
+		devicesConfig.Rtlfs.Devices = target.GetRtlfsDevices(c.Spec.Storage.Directories, &storeConfig)
+		deploymentConfig.DevConfig[node.Name] = devicesConfig
+	}
+
+	// Add Directories to deploymentConfig
+	clusterStorageConfig := config.ToStoreConfig(c.Spec.Storage.Config)
+	deploymentConfig.Directories = target.GetRtlfsDevices(c.Spec.Storage.Directories, &clusterStorageConfig)
+
+	if len(c.Spec.Storage.Directories) > 0 && (len(c.Spec.DataDirHostPath) > 0 || c.Spec.DataVolumeSize.Value() != 0) {
+		deploymentConfig.DeploymentType = edgefsv1alpha1.DeploymentRtlfs
+		deploymentConfig.TransportKey = "rtlfs"
+
+		// check directories devices count on all nodes
```

Comment:
Please keep all comments the same, that either the first character is capitalized or not (e.g., comment on line 134 has first char capitalized). That the first char is capitalized is my opinion the better "looking" style.